suite-native/module-device-settings/src/hooks/usePinAction.tsx

```diff
@@ -0,0 +1,144 @@
+import { useCallback, useEffect } from 'react';
+import { useDispatch, useSelector } from 'react-redux';
+
+import { RouteProp, useNavigation, useRoute } from '@react-navigation/native';
+
+import {
+    removeButtonRequests,
+    selectIsDeviceConnected,
+    selectSelectedDevice,
+} from '@suite-common/wallet-core';
+import { useAlert } from '@suite-native/alerts';
+import { EventType, analytics } from '@suite-native/analytics';
+import { requestPrioritizedDeviceAccess } from '@suite-native/device-mutex';
+import { Translation, TxKeyPath } from '@suite-native/intl';
+import {
+    DeviceSettingsStackParamList,
+    DeviceStackRoutes,
+    PinActionType,
+    StackNavigationProps,
+} from '@suite-native/navigation';
+import { useToast } from '@suite-native/toasts';
+import TrezorConnect from '@trezor/connect';
+
+type NavigationProp = StackNavigationProps<
+    DeviceSettingsStackParamList,
+    DeviceStackRoutes.DevicePinProtection
+>;
+
+type ActionConfig = {
+    param: boolean | undefined;
```

Comment:
nitpick, feel free to ignore it because it was there before and I know it's not something you changed, but the name of the param `param` is cryptic and it would be nice to rename it to something like `remove` or make it generic same as on desktop by passing all params with type from Connect `Parameters<typeof TrezorConnect.changePin>[0]`

Reply:
https://github.com/trezor/trezor-suite/pull/17689/commits/3d5164a30ec1bcf5f8517850406b222fe2619319 as the latter option would require changes also in the `actionConfigMap` I opted for the first one.